pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -2528,83 +2508,101 @@ func mkisofs(output string, dir string) error {
+		phyAdapter := phyIOAdapterList.LookupAdapter(name)
+		if phyAdapter == nil {
+			handleIBDelete(ctx, name)
 		}
 	}
-	// Any add or modify?
-	for _, configIb := range config.IoBundleList {
-		statusIb := status.LookupIoBundle(configIb.Type, configIb.Name)
-		if statusIb == nil {
-			handleIBCreate(ctx, configIb, status)
-		} else if statusIb.AssignmentGroup != configIb.AssignmentGroup ||
-			statusIb.PciLong != configIb.PciLong ||
-			statusIb.Ifname != configIb.Ifname ||
-			statusIb.Irq != configIb.Irq ||
-			statusIb.Ioports != configIb.Ioports ||
-			statusIb.Serial != configIb.Serial {
 
-			handleIBModify(ctx, *statusIb, configIb, status)
+	// Any add or modify?
+	for _, phyAdapter := range phyIOAdapterList.AdapterList {
+		ib := *types.IoBundleFromPhyAdapter(phyAdapter)
+		currentIbPtr := aa.LookupIoBundle(0, phyAdapter.Phylabel)
+		if currentIbPtr == nil {
+			log.Infof("handlePhysicalIOAdapterListCreateModify: Adapter %s "+
+				"added. %+v\n", phyAdapter.Phylabel, ib)
+			handleIBCreate(ctx, ib)
+		} else if currentIbPtr.HasAdapterChanged(phyAdapter) {
+			log.Infof("handlePhysicalIOAdapterListCreateModify: Adapter %s "+
+				"changed. Current: %+v, New: %+v\n", phyAdapter.Phylabel,
+				*currentIbPtr, ib)
+			handleIBModify(ctx, ib)
+		} else {
+			log.Infof("handlePhysicalIOAdapterListCreateModify: Adapter %s "+
+				"- No Change\n", phyAdapter.Phylabel)
 		}
 	}
-	status.Initialized = true
+	aa.Initialized = true
 	ctx.publishAssignableAdapters()
-	log.Infof("handleAAModify() done\n")
+	log.Infof("handlePhysicalIOAdapterListCreateModify() done\n")
 }
 
-func handleAADelete(ctxArg interface{}, status *types.AssignableAdapters) {
+func handlePhysicalIOAdapterListDelete(ctxArg interface{},
+	key string, value interface{}) {
 
+	phyAdapterList := cast.CastPhysicalIOAdapterList(value)
 	ctx := ctxArg.(*domainContext)
-	log.Infof("handleAADelete()\n")
-	for _, statusIb := range (*status).IoBundleList {
-		handleIBDelete(ctx, statusIb, status)
+	log.Infof("handlePhysicalIOAdapterListDelete: ALL PhysicalIoAdapters " +
+		"deleted\n")
+
+	for indx := range phyAdapterList.AdapterList {
+		name := phyAdapterList.AdapterList[indx].Phylabel
+		log.Infof("handlePhysicalIOAdapterListDelete: Deleting Adapter %s\n",
+			name)
+		handleIBDelete(ctx, name)
 	}
 	ctx.publishAssignableAdapters()
-	log.Infof("handleAADelete() done\n")
+	log.Infof("handlePhysicalIOAdapterListDelete done\n")
 }
 
 // Process new IoBundles. Check if PCI device exists, and check that not
 // used in a DevicePortConfig/DeviceNetworkStatus
 // Assign to pciback
-func handleIBCreate(ctx *domainContext, ib types.IoBundle,
-	status *types.AssignableAdapters) {
+func handleIBCreate(ctx *domainContext, ib types.IoBundle) {
 
 	log.Infof("handleIBCreate(%d %s %s)", ib.Type, ib.Name, ib.AssignmentGroup)
-	if err := checkAndSetIoBundle(ctx, &ib); err != nil {
+	aa := ctx.assignableAdapters
+	if err := checkAndSetIoBundle(ctx, &ib, false); err != nil {
```

Comment:
Shouldn't this set publish=true?

Reply:
No.. the caller of handleIBCreate is actually publishing the Adapters..

Reply:
OK